Automated Smoketest: :x:
Sha: d60c8e0fec049bc44dc0bea50bc429c5df7842b9
Empty table
eclwatch/DynamicESDLMethodWidget.js
eclwatch/LFDetailsWidget.js

Build: failed
Number of Error(s): 8
[13:14:46] Error in plugin gulp-jshint
[  0%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/client/CoreErrors.cpp.o
[100%] Building CXX object aws-cpp-sdk-sqs/CMakeFiles/aws-cpp-sdk-sqs.dir/source/SQSErrors.cpp.o
/mnt/disk1/home/vamosax/smoketest/PR-10613/HPCC-Platform/testing/unittests/dalitests.cpp:1140:13: error: looser throw specifier for \xe2\x80\x98virtual CDaliSDSTests::testSDSSubs2()::CSubscriberContainer::~CSubscriberContainer() noexcept (false)\xe2\x80\x99
/mnt/disk1/home/vamosax/smoketest/PR-10613/HPCC-Platform/testing/unittests/./../../system/jlib/jiface.hpp:102:17: error:   overriding \xe2\x80\x98virtual CSimpleInterface::~CSimpleInterface() noexcept (true)\xe2\x80\x99
make[2]: *** [testing/unittests/CMakeFiles/unittests.dir/dalitests.cpp.o] Error 1
make[1]: *** [testing/unittests/CMakeFiles/unittests.dir/all] Error 2
make: *** [all] Error 2
Get ECLWatch build result

--- a/testing/unittests/dalitests.cpp
+++ b/testing/unittests/dalitests.cpp
@@ -1107,7 +1107,7 @@ public:
     }
     void testSDSSubs2()
     {
-        class CSubscriberContainer : public CSimpleInterface
+        class CSubscriberContainer : public CInterface
         {
             class CSubscriber : public CSimpleInterfaceOf<ISDSSubscription>
             {
@@ -1137,7 +1137,7 @@ public:
                 id = querySDS().subscribe(xpath, *subscriber, sub, !sub);
                 PROGLOG("Subscribed to %s", xpath);
             }
-            ~CSubscriberContainer()
+            virtual void beforeDispose() override
             {
                 querySDS().unsubscribe(id);
             }

